tests/destructor/t16607.nim

```diff
@@ -0,0 +1,22 @@
+discard """
+  matrix: "--gc:refc; --gc:arc"
+"""
+
```

Comment:
in followup PR please add
```nim
# bug #16607
```

see rationale here https://github.com/nim-lang/Nim/pull/17316/files#r591062106